lib/feedsme/index.js

```diff
@@ -535,10 +540,12 @@ Feedsme.prototype.increment = function inc({ publish, strategy, payload, head, p
     // We need to figure out the latest head version in case a rollback happened
     // which would mess up our whole system here. If our version fetched is
     // somehow greater than the current build head, we will increrment that
-    // version
+    // version only if its in the same major version line
     //
     const latestHead = this._latestHead(head);
-    if (!latest.version || semver.lte(latest.version, latestHead)) {
+    if (!latest.version ||
+      (semver.lte(latest.version, latestHead) // head > latest.version
+        && semver.major(latest.version) === semver.major(latestHead))) { // && same major version line
```

Comment:
This logic change could use a test case

Reply:
will make an issue for it and defer for now